In [57]:

import pandas as pd
import numpy as np
#df=pd.read_excel('ODI_cricket.xlsx', sheet_name='bowler', engine='openpyxl')
#display(df.head())

In [58]:
from google.colab import files
uploaded= files.upload()

Saving ODI_cricket1.xlsx to ODI_cricket1 (1).xlsx


# **Displaying the first 10 rows of the data frame**

In [59]:
import io
df = pd.read_excel(io.BytesIO(uploaded['ODI_cricket1.xlsx']))
#df = pd.read_excel(io.BytesIO(uploaded['ODI_cricket1.xlsx', sheet_name='bowler',engine='openpyxl']))
display(df.head(10))

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1             Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2            Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3          WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
5       SM Pollock (Afr/ICC/SA)  1996-2008  303   297  15712   9631   393   
6          GD McGrath (AUS/ICC)  1993-2007  250   248  12970   8391   381   
7                   B Lee (AUS)  2000-2012  221   217  11185   8877   380   
8               SL Malinga (SL)  2004-2019  226   220  10936   9760   338   
9         A Kumble (Asia/INDIA)  1990-2007  271   265  14496  10412   337   

     Ave  Econ    SR   4   5  
0  23.08  3.93  35.2  15  10  
1  23.52  3.89  36.2  17   6  
2  23.84  4.68  30.5  14  13  
3  27.53  4.18  39.4   9   4  
4  34.51  4.62  44.7   4   9  
5  24.50  3.67  39.9  12   5  
6  22.02  3.88  34.0   9   7  
7  23.36  4.76  29.4  14   9  
8  28.87  5.35  32.3  11   8  
9  30.89  4.30  43.0   8   2

# **Markdown Cell for explaining the meaning of each column**
 


*   **Span:** Total period of time playing a player.
*   **Mat:** The Number of match played.
*   **Inns:** The number of innings in which the bowler actually bowled.
*   **Balls:** The number of balls bowled.
*   **Runs:** Total number of runs conceded.
*   **Wkts:** The number of wickets taken.
*   **Ave:** The number of runs a bowler has conceded per wicket taken.
*   **Econ:** The average number of runs conceded per over. (Econ = Runs/Overs bowled).

*   **SR:** Strike rate. The average number of balls bowled per wicket taken. (SR = Balls/W)
*   **4:** The number of innings in which the bowler took at least four wickets.
*  **5:** The number of innings in which the bowler took at least five wickets.




# **Finding the number of rows and columns in the dataframe**

In [60]:
print("Total number of rows=", df.shape[0])
print("Total number of columns=", df.shape[1])

Total number of rows= 77
Total number of columns= 12


# **Finding the data statistics**

In [62]:
display(df.describe())

Mat        Inns         Balls          Runs        Wkts  \
count   77.000000   77.000000     77.000000     77.000000   77.000000   
mean   194.402597  181.194805   8839.402597   6671.714286  233.805195   
std     82.485606   67.958393   3316.055457   2245.839029   84.406603   
min     80.000000   76.000000   4074.000000   2821.000000  151.000000   
25%    136.000000  128.000000   6182.000000   5058.000000  173.000000   
50%    170.000000  164.000000   8054.000000   6192.000000  199.000000   
75%    227.000000  218.000000  10750.000000   8021.000000  272.000000   
max    463.000000  372.000000  18811.000000  13632.000000  534.000000   

             Ave       Econ         SR          4          5  
count  77.000000  77.000000  77.000000  77.000000  77.000000  
mean   28.958052   4.596753  37.909091   6.350649   2.870130  
std     4.826768   0.515814   6.060901   3.556929   2.530606  
min    18.680000   3.300000  26.100000   1.000000   0.000000  
25%    24.970000   4.280000  33.000000   3.000000   1.000000  
50%    29.290000   4.660000  37.800000   6.000000   2.000000  
75%    31.900000   4.920000  41.400000   8.000000   4.000000  
max    44.480000   5.830000  52.500000  17.000000  13.000000

# Checking data type & missing values of each columns

In [63]:
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  77 non-null     object 
 1   Span    77 non-null     object 
 2   Mat     77 non-null     int64  
 3   Inns    77 non-null     int64  
 4   Balls   77 non-null     int64  
 5   Runs    77 non-null     int64  
 6   Wkts    77 non-null     int64  
 7   Ave     77 non-null     float64
 8   Econ    77 non-null     float64
 9   SR      77 non-null     float64
 10  4       77 non-null     int64  
 11  5       77 non-null     int64  
dtypes: float64(3), int64(7), object(2)
memory usage: 7.3+ KB


None

AS all the columns(12) shows non-null with respect to existing all rows(77), So there is no missing values present in the dataset.  

# **Displaying existing all columns names**

In [64]:
display(df.columns)

Index(['Player',   'Span',    'Mat',   'Inns',  'Balls',   'Runs',   'Wkts',
          'Ave',   'Econ',     'SR',        4,        5],
      dtype='object')

# **Renaming the column names**

In [65]:
df = df.rename(columns={'Mat':'Match', 
                        'Inns':'Innings',
                        #'Balls': 'The number of balls bowled',
                        #'Runs': 'The number of runs conceded',
                        'Wkts': 'Wickets',
                        #'Wkts':'The_number_of_wikets_taken',
                        'Ave': 'Bowling_average',
                        'Econ':'Economy_rate',
                        'SR': 'Strike_Rate'})

display(df.head())

Player       Span  Match  Innings  Balls   Runs  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011    350      341  18811  12326   
1             Wasim Akram (PAK)  1984-2003    356      351  18186  11812   
2            Waqar Younis (PAK)  1989-2003    262      258  12698   9919   
3          WPUJC Vaas (Asia/SL)  1994-2008    322      320  15775  11014   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015    398      372  17670  13632   

   Wickets  Bowling_average  Economy_rate  Strike_Rate   4   5  
0      534            23.08          3.93         35.2  15  10  
1      502            23.52          3.89         36.2  17   6  
2      416            23.84          4.68         30.5  14  13  
3      400            27.53          4.18         39.4   9   4  
4      395            34.51          4.62         44.7   4   9

# **Spliting the player column into two different columns named, player_Name & Country.**

In [66]:
df[["Player_Name", "Country"]] = df['Player'].str.split("(", expand=True)

display(df.head())

Player       Span  Match  Innings  Balls   Runs  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011    350      341  18811  12326   
1             Wasim Akram (PAK)  1984-2003    356      351  18186  11812   
2            Waqar Younis (PAK)  1989-2003    262      258  12698   9919   
3          WPUJC Vaas (Asia/SL)  1994-2008    322      320  15775  11014   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015    398      372  17670  13632   

   Wickets  Bowling_average  Economy_rate  Strike_Rate   4   5  \
0      534            23.08          3.93         35.2  15  10   
1      502            23.52          3.89         36.2  17   6   
2      416            23.84          4.68         30.5  14  13   
3      400            27.53          4.18         39.4   9   4   
4      395            34.51          4.62         44.7   4   9   

       Player_Name        Country  
0  M Muralitharan    Asia/ICC/SL)  
1     Wasim Akram            PAK)  
2    Waqar Younis            PAK)  
3      WPUJC Vaas        Asia/SL)  
4   Shahid Afridi   Asia/ICC/PAK)

# **Omitting the parentheses of the last part of first bracket. i.e.  )**

In [68]:
df['Country'] = df['Country'].str.replace(")", "")

display(df.head())

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  """Entry point for launching an IPython kernel.


Player       Span  Match  Innings  Balls   Runs  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011    350      341  18811  12326   
1             Wasim Akram (PAK)  1984-2003    356      351  18186  11812   
2            Waqar Younis (PAK)  1989-2003    262      258  12698   9919   
3          WPUJC Vaas (Asia/SL)  1994-2008    322      320  15775  11014   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015    398      372  17670  13632   

   Wickets  Bowling_average  Economy_rate  Strike_Rate   4   5  \
0      534            23.08          3.93         35.2  15  10   
1      502            23.52          3.89         36.2  17   6   
2      416            23.84          4.68         30.5  14  13   
3      400            27.53          4.18         39.4   9   4   
4      395            34.51          4.62         44.7   4   9   

       Player_Name       Country  
0  M Muralitharan    Asia/ICC/SL  
1     Wasim Akram            PAK  
2    Waqar Younis            PAK  
3      WPUJC Vaas        Asia/SL  
4   Shahid Afridi   Asia/ICC/PAK

In [69]:
df['Country'].value_counts()

INDIA           9
PAK             9
AUS             8
NZ              6
SL              5
WI              5
Asia/INDIA      4
ENG             4
Asia/SL         3
SA              3
Afr/ICC/SA      2
AUS/ICC         2
Asia/ICC/PAK    2
ENG/ICC         2
ICC/NZ          2
BAN             2
Afr/SA          2
ICC/WI          1
Asia/ICC/SL     1
Afr/ZIM         1
ICC/SA          1
Asia/PAK        1
Asia/BAN        1
AFG             1
Name: Country, dtype: int64

In [70]:
def icc_check(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"

In [71]:
df['played_for_ICC'] = df['Country'].apply(icc_check)
display(df.head())


Player       Span  Match  Innings  Balls   Runs  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011    350      341  18811  12326   
1             Wasim Akram (PAK)  1984-2003    356      351  18186  11812   
2            Waqar Younis (PAK)  1989-2003    262      258  12698   9919   
3          WPUJC Vaas (Asia/SL)  1994-2008    322      320  15775  11014   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015    398      372  17670  13632   

   Wickets  Bowling_average  Economy_rate  Strike_Rate   4   5  \
0      534            23.08          3.93         35.2  15  10   
1      502            23.52          3.89         36.2  17   6   
2      416            23.84          4.68         30.5  14  13   
3      400            27.53          4.18         39.4   9   4   
4      395            34.51          4.62         44.7   4   9   

       Player_Name       Country played_for_ICC  
0  M Muralitharan    Asia/ICC/SL            Yes  
1     Wasim Akram            PAK             No  
2    Waqar Younis            PAK             No  
3      WPUJC Vaas        Asia/SL             No  
4   Shahid Afridi   Asia/ICC/PAK            Yes

In [72]:
#display(df.head(37))
#display(df.tail(40))
display(df.sample(8))

Player       Span  Match  Innings  Balls   Runs  Wickets  \
50         Imran Khan (PAK)  1974-1992    175      153   7461   4844      182   
29      HH Streak (Afr/ZIM)  1993-2005    189      185   9468   7129      239   
47        RA Jadeja (INDIA)  2009-2020    168      164   8557   7024      188   
27            KD Mills (NZ)  2001-2015    170      169   8230   6485      240   
22  Abdul Razzaq (Asia/PAK)  1996-2011    265      254  10941   8564      269   
58        IK Pathan (INDIA)  2004-2012    120      118   5855   5142      173   
34       CJ McDermott (AUS)  1985-1996    138      138   7461   5018      203   
1         Wasim Akram (PAK)  1984-2003    356      351  18186  11812      502   

    Bowling_average  Economy_rate  Strike_Rate   4  5    Player_Name  \
50            26.61          3.89         40.9   3  1    Imran Khan    
29            29.82          4.51         39.6   7  1     HH Streak    
47            37.36          4.92         45.5   7  1     RA Jadeja    
27            27.02          4.72         34.2   8  1      KD Mills    
22            31.83          4.69         40.6   8  3  Abdul Razzaq    
58            29.72          5.26         33.8   5  2     IK Pathan    
34            24.71          4.03         36.7   4  1  CJ McDermott    
1             23.52          3.89         36.2  17  6   Wasim Akram    

     Country played_for_ICC  
50       PAK             No  
29   Afr/ZIM             No  
47     INDIA             No  
27        NZ             No  
22  Asia/PAK             No  
58     INDIA             No  
34       AUS             No  
1        PAK             No

# **Finding the number of players played for ICC**

In [73]:
print(df['played_for_ICC'].value_counts())

No     64
Yes    13
Name: played_for_ICC, dtype: int64


**Here, Thirteen players played for ICC**

In [74]:
df['Country'] = df['Country'].str.replace("ICC/", "")
df['Country'] = df['Country'].str.replace("/ICC", "")
df['Country'] = df['Country'].str.replace("Asia/", "")
df['Country'] = df['Country'].str.replace("Afr/", "")
display(df.head(10))
#display(df.head(37))
#display(df.tail(40))

Player       Span  Match  Innings  Balls   Runs  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011    350      341  18811  12326   
1             Wasim Akram (PAK)  1984-2003    356      351  18186  11812   
2            Waqar Younis (PAK)  1989-2003    262      258  12698   9919   
3          WPUJC Vaas (Asia/SL)  1994-2008    322      320  15775  11014   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015    398      372  17670  13632   
5       SM Pollock (Afr/ICC/SA)  1996-2008    303      297  15712   9631   
6          GD McGrath (AUS/ICC)  1993-2007    250      248  12970   8391   
7                   B Lee (AUS)  2000-2012    221      217  11185   8877   
8               SL Malinga (SL)  2004-2019    226      220  10936   9760   
9         A Kumble (Asia/INDIA)  1990-2007    271      265  14496  10412   

   Wickets  Bowling_average  Economy_rate  Strike_Rate   4   5  \
0      534            23.08          3.93         35.2  15  10   
1      502            23.52          3.89         36.2  17   6   
2      416            23.84          4.68         30.5  14  13   
3      400            27.53          4.18         39.4   9   4   
4      395            34.51          4.62         44.7   4   9   
5      393            24.50          3.67         39.9  12   5   
6      381            22.02          3.88         34.0   9   7   
7      380            23.36          4.76         29.4  14   9   
8      338            28.87          5.35         32.3  11   8   
9      337            30.89          4.30         43.0   8   2   

       Player_Name Country played_for_ICC  
0  M Muralitharan       SL            Yes  
1     Wasim Akram      PAK             No  
2    Waqar Younis      PAK             No  
3      WPUJC Vaas       SL             No  
4   Shahid Afridi      PAK            Yes  
5      SM Pollock       SA            Yes  
6      GD McGrath      AUS            Yes  
7           B Lee      AUS             No  
8      SL Malinga       SL             No  
9        A Kumble    INDIA             No

# **Finding the total number of different countries present in the dataset.**

In [75]:
df['Country'].value_counts()

INDIA    13
PAK      12
AUS      10
SL        9
SA        8
NZ        8
ENG       6
WI        6
BAN       3
ZIM       1
AFG       1
Name: Country, dtype: int64

**Here eleven different countries present in the dataset.**

*   **There are ten Australian Bowlers Present in this dataset**
*   **Also, There are three Bangladeshi players Present in this dataset**



# **Spliting the span column into two different columns named, start_year end_year and changing the data type string to integer.**

In [76]:
df[['start_year', 'end_year']] = df['Span'].str.split("-", expand=True)
df['start_year'] = df['start_year'].astype('int') 
df['end_year'] = df['end_year'].astype('int')
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Player           77 non-null     object 
 1   Span             77 non-null     object 
 2   Match            77 non-null     int64  
 3   Innings          77 non-null     int64  
 4   Balls            77 non-null     int64  
 5   Runs             77 non-null     int64  
 6   Wickets          77 non-null     int64  
 7   Bowling_average  77 non-null     float64
 8   Economy_rate     77 non-null     float64
 9   Strike_Rate      77 non-null     float64
 10  4                77 non-null     int64  
 11  5                77 non-null     int64  
 12  Player_Name      77 non-null     object 
 13  Country          77 non-null     object 
 14  played_for_ICC   77 non-null     object 
 15  start_year       77 non-null     int64  
 16  end_year         77 non-null     int64  
dtypes: float64(3), int

# **Finding the player who played for the longest period of time**

In [77]:
df['years_played'] = df['end_year'] - df['start_year']+1
#display(df.head(37))
#display(df.tail(40))
df.sort_values(by='years_played', ascending = False).head(8)
#display(df.tail(10))


Player       Span  Match  Innings  Balls   Runs  \
72          SR Tendulkar (INDIA)  1989-2012    463      270   8054   6850   
10       ST Jayasuriya (Asia/SL)  1989-2011    445      368  14874  11871   
66            Shoaib Malik (PAK)  1999-2019    287      217   7958   6192   
62             CH Gayle (ICC/WI)  1999-2019    301      199   7424   5926   
4   Shahid Afridi (Asia/ICC/PAK)  1996-2015    398      372  17670  13632   
20   Mashrafe Mortaza (Asia/BAN)  2001-2020    220      220  10922   8893   
1              Wasim Akram (PAK)  1984-2003    356      351  18186  11812   
0   M Muralitharan (Asia/ICC/SL)  1993-2011    350      341  18811  12326   

    Wickets  Bowling_average  Economy_rate  Strike_Rate   4   5  \
72      154            44.48          5.10         52.2   4   2   
10      323            36.75          4.78         46.0   8   4   
66      158            39.18          4.66         50.3   1   0   
62      167            35.48          4.78         44.4   3   1   
4       395            34.51          4.62         44.7   4   9   
20      270            32.93          4.88         40.4   7   1   
1       502            23.52          3.89         36.2  17   6   
0       534            23.08          3.93         35.2  15  10   

          Player_Name Country played_for_ICC  start_year  end_year  \
72      SR Tendulkar    INDIA             No        1989      2012   
10     ST Jayasuriya       SL             No        1989      2011   
66      Shoaib Malik      PAK             No        1999      2019   
62          CH Gayle       WI            Yes        1999      2019   
4      Shahid Afridi      PAK            Yes        1996      2015   
20  Mashrafe Mortaza      BAN             No        2001      2020   
1        Wasim Akram      PAK             No        1984      2003   
0     M Muralitharan       SL            Yes        1993      2011   

    years_played  
72            24  
10            23  
66            21  
62            21  
4             20  
20            20  
1             20  
0             19

**SR Tendulkar played for the longest period of time, about 24 years**

# **Finding the players who played lowest period of time**

In [78]:
df.sort_values(by='years_played', ascending = True).head(8)

Player       Span  Match  Innings  Balls  Runs  Wickets  \
40     BKV Prasad (INDIA)  1994-2001    161      160   8129  6332      196   
74      Rashid Khan (AFG)  2015-2022     80       76   4074  2821      151   
73        BAW Mendis (SL)  2008-2015     87       84   4154  3324      152   
49      Saeed Ajmal (PAK)  2008-2015    113      112   6000  4182      184   
44        L Klusener (SA)  1996-2004    171      164   7336  5751      192   
56       Imran Tahir (SA)  2011-2019    107      104   5541  4297      173   
14  Saqlain Mushtaq (PAK)  1995-2003    169      165   8770  6275      288   
55       NW Bracken (AUS)  2001-2009    116      116   5759  4240      174   

    Bowling_average  Economy_rate  Strike_Rate   4  5       Player_Name  \
40            32.30          4.67         41.4   3  1       BKV Prasad    
74            18.68          4.15         26.9   5  4      Rashid Khan    
73            21.86          4.80         27.3   7  3       BAW Mendis    
49            22.72          4.18         32.6   6  2      Saeed Ajmal    
44            29.95          4.70         38.2   1  6       L Klusener    
56            24.83          4.65         32.0   7  3      Imran Tahir    
14            21.78          4.29         30.4  11  6  Saqlain Mushtaq    
55            24.36          4.41         33.0   5  2       NW Bracken    

   Country played_for_ICC  start_year  end_year  years_played  
40   INDIA             No        1994      2001             8  
74     AFG             No        2015      2022             8  
73      SL             No        2008      2015             8  
49     PAK             No        2008      2015             8  
44      SA             No        1996      2004             9  
56      SA             No        2011      2019             9  
14     PAK             No        1995      2003             9  
55     AUS             No        2001      2009             9

**Here, BKV Prasad (INDIA), Rashid Khan (AFG), BAW Mendis (SL), and Saeed Ajmal (PAK) had played for the lowest period of time, about 8 years.**

# **Finding the player who had lowest economy rate**

In [79]:
df.sort_values(by='Economy_rate', ascending =True).head(8)

Player       Span  Match  Innings  Balls   Runs  Wickets  \
65           RJ Hadlee (NZ)  1973-1990    115      112   6182   3407      158   
32         CEL Ambrose (WI)  1988-2000    176      175   9353   5429      225   
67         MD Marshall (WI)  1980-1992    136      134   7175   4233      157   
5   SM Pollock (Afr/ICC/SA)  1996-2008    303      297  15712   9631      393   
25      N Kapil Dev (INDIA)  1978-1994    225      221  11202   6945      253   
31            CA Walsh (WI)  1985-2000    205      204  10822   6918      227   
6      GD McGrath (AUS/ICC)  1993-2007    250      248  12970   8391      381   
1         Wasim Akram (PAK)  1984-2003    356      351  18186  11812      502   

    Bowling_average  Economy_rate  Strike_Rate   4  5   Player_Name Country  \
65            21.56          3.30         39.1   1  5    RJ Hadlee       NZ   
32            24.12          3.48         41.5   6  4  CEL Ambrose       WI   
67            26.96          3.53         45.7   6  0  MD Marshall       WI   
5             24.50          3.67         39.9  12  5   SM Pollock       SA   
25            27.45          3.71         44.2   3  1  N Kapil Dev    INDIA   
31            30.47          3.83         47.6   6  1     CA Walsh       WI   
6             22.02          3.88         34.0   9  7   GD McGrath      AUS   
1             23.52          3.89         36.2  17  6  Wasim Akram      PAK   

   played_for_ICC  start_year  end_year  years_played  
65             No        1973      1990            18  
32             No        1988      2000            13  
67             No        1980      1992            13  
5             Yes        1996      2008            13  
25             No        1978      1994            17  
31             No        1985      2000            16  
6             Yes        1993      2007            15  
1              No        1984      2003            20

**Here, RJ Hadlee (NZ) had the lowest economy rate, about 3.30**

# **Finding the player who had lowest strike rate**

In [80]:
df.sort_values(by='Strike_Rate', ascending =True).head(8)

Player       Span  Match  Innings  Balls  Runs  Wickets  \
41         MA Starc (AUS)  2010-2021     99       99   5099  4379      195   
74      Rashid Khan (AFG)  2015-2022     80       76   4074  2821      151   
73        BAW Mendis (SL)  2008-2015     87       84   4154  3324      152   
7             B Lee (AUS)  2000-2012    221      217  11185  8877      380   
60          TA Boult (NZ)  2012-2021     93       93   5117  4261      169   
14  Saqlain Mushtaq (PAK)  1995-2003    169      165   8770  6275      288   
2      Waqar Younis (PAK)  1989-2003    262      258  12698  9919      416   
46      M Morkel (Afr/SA)  2007-2018    117      114   5760  4761      188   

    Bowling_average  Economy_rate  Strike_Rate   4   5       Player_Name  \
41            22.45          5.15         26.1  11   8         MA Starc    
74            18.68          4.15         26.9   5   4      Rashid Khan    
73            21.86          4.80         27.3   7   3       BAW Mendis    
7             23.36          4.76         29.4  14   9            B Lee    
60            25.21          4.99         30.2   8   5         TA Boult    
14            21.78          4.29         30.4  11   6  Saqlain Mushtaq    
2             23.84          4.68         30.5  14  13     Waqar Younis    
46            25.32          4.95         30.6   7   2         M Morkel    

   Country played_for_ICC  start_year  end_year  years_played  
41     AUS             No        2010      2021            12  
74     AFG             No        2015      2022             8  
73      SL             No        2008      2015             8  
7      AUS             No        2000      2012            13  
60      NZ             No        2012      2021            10  
14     PAK             No        1995      2003             9  
2      PAK             No        1989      2003            15  
46      SA             No        2007      2018            12

Here, MA Starc (AUS) had lowest strike rate, about 26.1

# **Finding the player who had the lowest bowling average**

In [81]:
df.sort_values(by='Bowling_average', ascending =True).head(8)

Player       Span  Match  Innings  Balls  Runs  Wickets  \
74      Rashid Khan (AFG)  2015-2022     80       76   4074  2821      151   
65         RJ Hadlee (NZ)  1973-1990    115      112   6182  3407      158   
19         AA Donald (SA)  1991-2003    164      162   8561  5926      272   
14  Saqlain Mushtaq (PAK)  1995-2003    169      165   8770  6275      288   
73        BAW Mendis (SL)  2008-2015     87       84   4154  3324      152   
6    GD McGrath (AUS/ICC)  1993-2007    250      248  12970  8391      381   
41         MA Starc (AUS)  2010-2021     99       99   5099  4379      195   
49      Saeed Ajmal (PAK)  2008-2015    113      112   6000  4182      184   

    Bowling_average  Economy_rate  Strike_Rate   4  5       Player_Name  \
74            18.68          4.15         26.9   5  4      Rashid Khan    
65            21.56          3.30         39.1   1  5        RJ Hadlee    
19            21.78          4.15         31.4  11  2        AA Donald    
14            21.78          4.29         30.4  11  6  Saqlain Mushtaq    
73            21.86          4.80         27.3   7  3       BAW Mendis    
6             22.02          3.88         34.0   9  7       GD McGrath    
41            22.45          5.15         26.1  11  8         MA Starc    
49            22.72          4.18         32.6   6  2      Saeed Ajmal    

   Country played_for_ICC  start_year  end_year  years_played  
74     AFG             No        2015      2022             8  
65      NZ             No        1973      1990            18  
19      SA             No        1991      2003            13  
14     PAK             No        1995      2003             9  
73      SL             No        2008      2015             8  
6      AUS            Yes        1993      2007            15  
41     AUS             No        2010      2021            12  
49     PAK             No        2008      2015             8

**Here, Rashid Khan (AFG) had the lowest bowling average, about 18.68**

# **Removing Unnessesary Columns**

In [85]:
#df.drop(['Player', 'Span'], axis=1)
df = df.drop('Player', axis=1)
display(df.head(8))

Span  Match  Innings  Balls   Runs  Wickets  Bowling_average  \
0  1993-2011    350      341  18811  12326      534            23.08   
1  1984-2003    356      351  18186  11812      502            23.52   
2  1989-2003    262      258  12698   9919      416            23.84   
3  1994-2008    322      320  15775  11014      400            27.53   
4  1996-2015    398      372  17670  13632      395            34.51   
5  1996-2008    303      297  15712   9631      393            24.50   
6  1993-2007    250      248  12970   8391      381            22.02   
7  2000-2012    221      217  11185   8877      380            23.36   

   Economy_rate  Strike_Rate   4   5      Player_Name Country played_for_ICC  \
0          3.93         35.2  15  10  M Muralitharan       SL            Yes   
1          3.89         36.2  17   6     Wasim Akram      PAK             No   
2          4.68         30.5  14  13    Waqar Younis      PAK             No   
3          4.18         39.4   9   4      WPUJC Vaas       SL             No   
4          4.62         44.7   4   9   Shahid Afridi      PAK            Yes   
5          3.67         39.9  12   5      SM Pollock       SA            Yes   
6          3.88         34.0   9   7      GD McGrath      AUS            Yes   
7          4.76         29.4  14   9           B Lee      AUS             No   

   start_year  end_year  years_played  
0        1993      2011            19  
1        1984      2003            20  
2        1989      2003            15  
3        1994      2008            15  
4        1996      2015            20  
5        1996      2008            13  
6        1993      2007            15  
7        2000      2012            13

In [89]:
#new_col_sequence = ['Player_Name', 'Country', 'Span', 'Match', 'Innings', 'Balls', 'Runs', 'Wickts', 'Bowling_average', 'Economy_rate', 'Strike_Rate', '4', '5' ]
#df = df['new_col_sequence']
#display(df.head())